test/api/query.test.cpp

```diff
@@ -127,3 +130,121 @@ TEST(Query, QuerySourceFeaturesFilter) {
     EXPECT_EQ(features3.size(), 1u);
 }
 
+TEST(Query, QueryFeatureExtensions) {
```

Comment:
Typically unit tests are smaller and more focused. It would be nice to break this test into multiple smaller tests. At the least, it would help to separate the positive and negative codepaths.

Reply:
Thank you for looking. I will split test into few smaller unit tests.